### PR TITLE
ALL: Freezetag game mode

### PR DESCRIPTION
**Pull request by mgerhardy**
This is merging the freezetag mod of v1.2 into the current code base.

Unlike the previous version which replaced TDM, this does implement the freezetag mode as a dedicated game mode. In order to test the code a little bit without the need to modify the arena file, I've made every team type map also available in the ft game type (as previously freezetag replaced the team game mode, this shouldn't be a problem).

The assets in wop-data still need some work and must be moved to proper locations and maybe even renamed. I've not yet tested this with any modified asset. I've just checked, that the game mechanic works and the bots are trying to unfreeze their team mates (also needs more testing).

- [x] Update arena file with new `ft` type
- [ ] Test bots with the new game mode
- [ ] Test betty light colors
- [ ] Help screen and "menu/help/loadinghelp_freezetag";
- [x] Weapon mark shaders are using the snowmark shader now
- [ ] item_freeze entity def?
- [x] Test that late joining players will auto-freeze
- [ ] check the `FT_DEFAULTWEAPONSET` - might have changed since v1.2
- [ ] Bot chat commands? (https://github.com/PadWorld-Entertainment/worldofpadman/issues/113)

# Update/add assets
- [x] cgs.media.FreezeScreenShader = trap_R_RegisterShaderNoMip("gfx/screen/ice");
- [ ] ~~cgs.media.thawBarShader = trap_R_RegisterShaderNoMip("hud/thawbar");~~
- [x] cgs.media.freezeIconShader = trap_R_RegisterShader("icons/FREEZEicon");
- [x] cgs.media.snowMarkShader = trap_R_RegisterShader("gfx/damage/snow_mark");
- [x] cgs.media.thawIcon = trap_R_RegisterShader("icons/thawicon");
- [x] cgs.media.iceblockModel = trap_R_RegisterModel("models/iceblock");

### CODE REVIEW

File: code/game/bg_public.h [hunk 1]
```diff
@@ -136,6 +137,7 @@ typedef enum {
 #define GTN__GT_CTF "Capture The Lolly"
 #define GTN__GT_SPRAY "Spray Your Color Team"
 #define GTN__GT_BALLOON "BigBalloon"
+#define GTN__GT_FREEZETAG "FreezeTag"
```
Comment (mgerhardy) [inline code/game/bg_public.h]: Not sure about the real name here?

File: code/cgame/cg_draw.c [hunk 2]
```diff
@@ -2238,6 +2235,8 @@ static void CG_DrawWarmup(void) {
 			s = "Free For All";
 		} else if (cgs.gametype == GT_TEAM) {
 			s = "Team Deathmatch";
+		} else if (cgs.gametype == GT_FREEZETAG) {
+			s = "Freeze Tag";
```
Comment (mgerhardy) [inline code/cgame/cg_draw.c]: We should really use the `GTN__GT_FREEZETAG` define here - and for the others, too

File: code/game/bg_public.h (hunk 1 above)
Comment (kai-li-wop) [reply code/game/bg_public.h]: Google reports 63,300,000 results for 'freeze tag' and "only" 890,000 for 'freezetag', so I would prefer 'Freeze Tag' as the name. Also 'BigBalloon' should be 'Big Balloon' so it is more consistent with the other names.

File: code/cgame/cg_draw.c (hunk 2 above)
Comment (kai-li-wop) [reply code/cgame/cg_draw.c]: Hmm, I wonder why it is called "Team Deathmatch". It should be "Free For All Team" as far as I remember it correct. We wanted to get rid of the word "death" in the game due to USK 12+ label.

Comment (mgerhardy) [reply code/cgame/cg_draw.c]: It's weird that these kind of strings are duplicated all over the code base. We should use the #defines for the game type names.

**Comment by kai-li-wop** [comment]
Changed the shader path a little: cgs.media.FreezeScreenShader = trap_R_RegisterShaderNoMip("gfx/screen/ice");

**Comment by mgerhardy** [comment]
`gfx/screen/ice` is used now. Thanks

File: code/cgame/cg_draw.c (hunk 2 above)
Comment (mgerhardy) [reply code/cgame/cg_draw.c]: I've changed the name in `CG_DrawWarmup` - and also fixed "Team Deathmatch" in the `main` branch now.

**Comment by kai-li-wop** [comment]
Necessary code changes 
cgs.media.freezeIconShader = trap_R_RegisterShader("icons/hint_freeze");
cgs.media.snowMarkShader = trap_R_RegisterShader("gfx/damage/snow_mrk");
cgs.media.thawIcon = trap_R_RegisterShader("icons/hint_thaw");

Could not find the 'thawbar' in xmas mod files. Where is it located? It's not in hud folder.

**Comment by mgerhardy** [comment]
`thawbar` was unused and dead code - removed it now. Everything else is updated.

**Comment by kai-li-wop** [comment]
The overhead "wallhack" freeze icon is not displayed for teammates like in the original mod. The freeze icon is only displayed in the scores tab right now. It should be displayed above teammates' heads like all other "hint" icons like in LPS when enabled in the game options menu. I don't know if it needs to be activated via a command right now, but we should implement a menu entry plus a tooltip for it in the game options menu. Also, I would like to suggest replacing the teammate icon with the freeze icon over the head for the time a teammate is frozen. Displaying both icons on top of each other like in the original mod looks a bit strange.

shot0000

**Comment by kai-li-wop** [comment]
Menu entry text: Freeze Tag Icon
Tooltip text: Show an icon over frozen teammates, visible through walls, to help you find them.

**Pull request closed without merging**